lib/rig/discovery/discovery.ex

```diff
@@ -0,0 +1,36 @@
+defmodule Rig.Discovery do
+  @moduledoc """
+  Module responsible to start Peerage application if relevant.
+  Peerage will be auto-configured by values from environment variables.
+  """
+
+  use Rig.Config, [:discovery_type]
+  require Logger
+
+  def start do
+    conf = config()
+    use_discovery(conf.discovery_type)
+  end
+
+  defp use_discovery(nil) do
+    Logger.info("No discovery for cluster specified")
+  end
+
+  defp use_discovery("dns") do
+    conf = config()
+    Application.put_env(:peerage, :via, Peerage.Via.Dns)
+    Application.put_env(:peerage, :dns_name, conf.dns_name)
```

Comment:
I like this! Using configuration for `:rig` to configure `:peerage` behind the curtain is a nice touch.